micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/MnAdminUserConfigurationTest.kt

```diff
@@ -0,0 +1,137 @@
+            Assertions.assertEquals(FIRSTNAME_VALUE, configuration.adminUser.firstname)
+            Assertions.assertEquals(LASTNAME_VALUE, configuration.adminUser.lastname)
+
+            assertAdminUser(processEngine, configuration.adminUser.id)
+            assertAdminGroup(processEngine)
+            assertAdminGroupAuthorizations(processEngine)
+        }
+
+        @Test
+        fun adminUserExists() {
+            val properties: MutableMap<String, Any> = HashMap()
+            properties[USERID_PROPERTY] = USERID_VALUE
+            properties[PASSWORD_PROPERTY] = PASSWORD_VALUE
+            properties[FIRSTNAME_PROPERTY] = "different first"
+            properties[LASTNAME_PROPERTY] = "different last"
+
+            ApplicationContext.build(properties).start().use { context ->
+                val localProcessEngine = context.getBean(ProcessEngine::class.java)
+                val localConfiguration = context.getBean(Configuration::class.java)
+                val localMnAdminUserConfiguration = context.findBean(MnAdminUserConfiguration::class.java)
+
+                Assertions.assertTrue(localMnAdminUserConfiguration.isPresent)
+                Assertions.assertEquals(USERID_VALUE, localConfiguration.adminUser.id)
+                Assertions.assertEquals(PASSWORD_VALUE, localConfiguration.adminUser.password)
+                Assertions.assertEquals("different first", localConfiguration.adminUser.firstname)
+                Assertions.assertEquals("different last", localConfiguration.adminUser.lastname)
+
+                val adminUser = queryUser(localProcessEngine, localConfiguration.adminUser.id)
+                Assertions.assertNotNull(adminUser)
+                // check for unchanged values
+                Assertions.assertEquals(FIRSTNAME_VALUE, adminUser.firstName)
+                Assertions.assertEquals(LASTNAME_VALUE, adminUser.lastName)
+
+                assertAdminGroup(localProcessEngine)
+                assertAdminGroupAuthorizations(localProcessEngine)
+            }
+        }
+    }
+
+    companion object {
+        const val USERID_PROPERTY = "camunda.bpm.admin-user.id"
+        const val USERID_VALUE = "admin"
+        const val PASSWORD_PROPERTY = "camunda.bpm.admin-user.password"
+        const val PASSWORD_VALUE = "password"
+        const val FIRSTNAME_PROPERTY = "camunda.bpm.admin-user.firstname"
+        const val FIRSTNAME_VALUE = "Firstname"
+        const val LASTNAME_PROPERTY = "camunda.bpm.admin-user.lastname"
+        const val LASTNAME_VALUE = "Lastname"
+        const val EMAIL_PROPERTY = "camunda.bpm.admin-user.email"
+        const val EMAIL_VALUE = "firstname@lastname.com"
+
+        private fun queryUser(processEngine: ProcessEngine?, userId: String): User {
+            return processEngine!!.identityService.createUserQuery().userId(userId).singleResult()
+        }
+
+        private fun assertAdminUser(processEngine: ProcessEngine?, userId: String) {
```

Comment:
* please rename this method (and the two following) to include what they are asserting.
* remove "private" from all methods